### DISCUSSION TITLE
Mind if I cannibalise your code to make something else?

### DISCUSSION THREAD

**Discussion by Extraltodeus** [thygate/stable-diffusion-webui-depthmap-script#26] [General]
I'm asking because I don't know how much I intend to blatantly copy-paste bits of your code. My goal is to create simple depth maps to cut out the background and re-blend stuff through img2img all at once from txt2img.

I'm already able to do txt2img2img with a custom script so using pillow to turn a color into alpha is just the next step : https://github.com/Extraltodeus/temp/blob/main/txt2img2img2img.py

Since you made it so much easier to understand/implement midas in the webui. Thanks by the way! :)

Note : if I ever finish what I intend to do, I will (of course) credit your work :)

**Comment by thygate** [comment]
You're welcome, have at it ! =)

Sounds like a great idea and project, can't wait to see the results.

**Reply by Extraltodeus** [reply]
Thank you :)

**Comment by Extraltodeus** [comment]
You can test it already if you want!
Just add these two scripts into your scripts folder (you need midas just as with yours) :
https://github.com/Extraltodeus/temp/blob/main/waigen.py
https://github.com/Extraltodeus/temp/blob/main/simple_depthmap.py

Haven't made more than the code yet but the results are pretty cool!

**Comment by Extraltodeus** [comment]
[It's done](https://github.com/Extraltodeus/multi-subject-render)

![image](https://user-images.githubusercontent.com/15731540/203831916-15ac77c1-8ff3-4502-ba24-a887efa015d6.png)

**Reply by thygate** [reply]
impressive, was a good idea and application for depth maps, and thanks for the mention! :smiley:

**Reply by Extraltodeus** [reply]
Thanks and you're welcome! :)
